routers/user/getStudentCV.ts

```diff
@@ -0,0 +1,45 @@
+import { NextFunction, Request, Response } from "express";
+import { Roles } from "../../src/entities/types/Roles";
+import { ValidationError } from "../../utils/errorsHandler";
+import { StudentsRating } from "../../src/entities/studentsRating/studentsRating.entity";
+import { myDataSource } from "../../config/database.configuration";
+import { StudentStatus } from "../../src/entities/types/studentsData";
+import { UserPayloadData } from "../../utils/createTokens";
+import { DestructuringToStudentsCV } from "../../utils/destructuringToStudentsCV";
+
+type RequestAndPayloadUser = Request & UserPayloadData;
+
+export const getStudentCV = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    const { id, role } = req.user as RequestAndPayloadUser;
+
+    if (role !== Roles.HR && role !== Roles.STUDENT) throw new ValidationError('Access denied.', 401);
+
+    const userId = req.params.id;
+
+    const results = await myDataSource
+      .getRepository(StudentsRating)
+      .createQueryBuilder('studentsRating')
+      .leftJoinAndSelect('studentsRating.user', 'user')
+      .leftJoinAndSelect('studentsRating.studentsData', 'studentsData')
+      .where(`user.role = '${Roles.STUDENT}'`)
+      .andWhere(`studentsData.status = '${StudentStatus.AVAILABLE}'`)
+      .andWhere('studentsRating.id = :userId', { userId })
+      .orderBy('studentsData.lastName, studentsData.firstName')
+      .getOne();
+
+    if (!results) return res.json([]);
+
+    const data = {
```

Comment:
Why you destructing result and after that destructing property from result 2 times? I think you can take result as argument to DestructingToStudentsCS and there retrive data. For example create private method and move destruction there